File: library/src/main/java/com/qingmei2/library/di/module/ServiceModule.java
package com.qingmei2.library.di.module;

import com.qingmei2.library.http.service.UserInfoService;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import retrofit2.Retrofit;

/**
 * Created by QingMei on 2017/8/15.
 * desc:
 */
@Module
public class ServiceModule {

    @Singleton
    @Provides
    UserInfoService provideUserInfoService(Retrofit retrofit) {
        return retrofit.create(UserInfoService.class);
    }
}

File: rhine/src/main/java/com/qingmei2/rhine/di/ServiceModule.kt
package com.qingmei2.rhine.di

import com.qingmei2.rhine.http.service.ServiceManager
import com.qingmei2.rhine.http.service.UserInfoService

import org.kodein.di.Kodein
import org.kodein.di.generic.bind
import org.kodein.di.generic.instance
import org.kodein.di.generic.singleton
import retrofit2.Retrofit

const val SERVICE_MODULE_TAG = "serviceModule"

val serviceModule = Kodein.Module(SERVICE_MODULE_TAG) {

    bind<UserInfoService>() with singleton {
        instance<Retrofit>().create(UserInfoService::class.java)
    }

    bind<ServiceManager>() with singleton {
        ServiceManager(instance())
    }
}